hack/make-rules/test-cmd-util.sh

```diff
@@ -3731,5 +3739,23 @@ __EOF__
   output_message=$(! KUBECTL_PLUGINS_PATH=test/fixtures/pkg/kubectl/plugins/ kubectl plugin error 2>&1)
   kube::test::if_has_string "${output_message}" 'error: exit status 1'
 
+  #################
+  # Impersonation #
+  #################
+  output_message=$(! kubectl get pods "${secure_kube_flags[@]}" --as-group=foo 2>&1)
+  kube::test::if_has_string "${output_message}" 'without impersonating a user'
+
+  if kube::test::if_supports_resource "${csr}" ; then
+    # --as
+    kubectl create -f hack/testdata/csr.yml "${secure_kube_flags[@]}" --as=user1
+    kube::test::get_object_assert 'csr/foo' '{{.spec.username}}' 'user1'
```

Comment:
should get a single group of `system:authenticated` in this case (check number of groups and specific value)